### PR TITLE
[bug](prepared statement) fix double type value incorrect in legacy planner

### PR DESCRIPTION
**Pull request by xy720**
## Proposed changes

Issue Number: close #xxx

fix double type value incorrect.

### CODE REVIEW

**Comment by xy720** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/analysis/LiteralExpr.java [hunk 1]
```diff
@@ -358,6 +358,7 @@ public static LiteralExpr getLiteralByMysqlType(int mysqlType, boolean isUnsigne
                 break;
             case 5: // MYSQL_TYPE_DOUBLE
                 literalExpr = LiteralExpr.create("0", Type.DOUBLE);
+                literalExpr.setType(Type.DOUBLE);
```
Comment (zclllyybb) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/LiteralExpr.java:361]: if double should set it type, should we just change `DOUBLE` branch in `LiteralExpr::create`?

Comment (xy720) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/LiteralExpr.java:361]: LiteralExpr::create is used in many places, it's best not to change it.

**Review by lide-reed — APPROVED** [review]
LGTM

**Review by cambyzju — APPROVED** [review]

**Pull request merged**